### ISSUE TITLE
Networkx multigraph plot inconsistency through graphviz

### ISSUE THREAD

**Issue by aanastasiou** [pygraphviz/pygraphviz#208]
OK, this is a strange one and possibly shared between networkx and pygraphviz but more likely to belong to pygraphviz. 

The key use case here is displaying a multigraph. I suspect that `networkx.draw_networkx()` does plot multiple edges but because it does not have a way of drawing alternative or parallel paths, it plots them one on top of the other.

The alternative here is to "drop" to graphviz and do the visualisation via that.

The first "problem" here is how you do this:

1. Save to dot, do the visualisation manually:
    * This **does not** work:
        ```
         import networkx
         G = networkx.read_graphml("somegraph.graphml") #Used yEd here, will provide the graph further below

         networkx.nx_agraph.write_dot(G,"thetest.dot")
       ```

      The multiple edges between nodes are not even emmitted to the file.

    * For this to work , use the `write_dot` from `pydot` as:
       ```
           networkx.nx_pydot.write_dot(G,"thetest.dot")
       ```
       In this case, the multiple edges **are** emitted.

OK, so this works with that tiny little "problem" in there. Since graphviz is accessible through networkx let's try to do it via `networkx`.

This works:

```
import networkx

G = networkx.MultiDiGraph()
G.add_edge(1,2)
G.add_edge(1,2)
networkx.nx_agraph.view_pygraphviz(G, prog="neato")
```

**BUT!**, reading in the graphml file (confirm that it is a multidigraph as expected and that it contains all necessary data) and trying to plot it with: `networkx.nx_agraph.view_pygraphviz(G, edgelabel="label", prog="neato")`, fails to produce the multiple edges on the graphviz output.

The reason here seems to be exactly the same, the `AGraph` does not contain all edges of the original Multigraph.

The reason for this seems to be within `pygraphviz` (?)

`pygraphviz` s `AGraph` doesn't like strings as keys (ok) but it also doesn't like numbers :/
(If it did like numbers -in general- it would be easy to at least provide a mapping between strings to numbers).

Specifically, this works:

```
import pygraphviz

A = pygraphviz.AGraph(name="alpha", strict=False, directed=False)

A.add_edge(1,2)
A.add_edge(1,2)

assert len(A.edges())==2
```

This does not work:

```
import pygraphviz

A = pygraphviz.AGraph(name="alpha", strict=False, directed=False)
A.add_edge(1,2,key="alpha")
A.add_edge(1,2,key="beta")

assert len(A.edges())==2
```

This neither works with numbers but it is more subtle than that. If the numbers are small (?) and sequential (?) it works otherwise it does not:

This works:

```
import pygraphviz

A = pygraphviz.AGraph(name="alpha", strict=False, directed=False)
A.add_edge(1,2,key=1)
A.add_edge(1,2,key=2)

assert len(A.edges())==2
```

This **does not** work :(

```
import pygraphviz

A = pygraphviz.AGraph(name="alpha", strict=False, directed=False)
A.add_edge(1,2,key=120)
A.add_edge(1,2,key=121)

assert len(A.edges())==2
```

This last snippet shows only one edge.

The reason I am focusing on `AGraph` here is because whatever a given `G` is (a generic networkx `G` a user may wish to display) it is [converted to AGraph](https://github.com/networkx/networkx/blob/master/networkx/drawing/nx_agraph.py#L382). 

The `G` is recognised succesfully, it is iterated succesfully (with all of its edges, their end points and keys), all of this data is passed to `AGraph` (through `add_edge`), **but!!!**, `AGraph` indexing on the `key` fails. Consequently, the `AGraph` does not contain all information that is passed to graphviz and the output that is produced is inconsistent with the underlying graph.

I am not sure how to correct this but it might be better addressed at the level of pygraphviz. Networkx provides any type of key for alternative edges (which is great). When the graph is initialised from a graphml file (perhaps from a yEd graphml file (?)), the edge keys follow the pattern `e0, e1, e2,....`.

My setup is as follows:

1. [Graphml test graph created via yEd](https://pastebin.com/7HNdcFaC).
2. Networkx 2.4
3. pygraphviz 1.5
4. Python 3.6.9
5. Graphviz version 2.40.1
6. OS: Linux

**Comment by dschult** [comment]
Fortunately or not, this is a "known bug" in Graphviz 2.40.x
It has been fixed in 2.42.x.
We should probably start requiring Graphviz 2.42 if only for this reason;

**Comment by dschult** [comment]
Yes, that is the right place, but No, not all the packages have2.42
available even though it was released last summer.  So, you’ll have to
check the version for your package manager.
The 2.38 version also works with multi graphs. The bug was introduced in
2.40.  We’ve been requiring 2.38 to avoid that bug until 2.42 came out.
The odd numbers are dev versions and also work.
The bug is in the agraph library of GraphViz, so it’s not just in the last
few calls. It’s right at the adding edges part. If you can get the edge in
there, it works fine.

Try v2.42 or if not available with your package mgr then try 2.38

**Comment by aanastasiou** [comment]
@dschult Thanks, that's great. It is not so much the version of the package manager as making sure I download the right version. Out of the box compilations do not include [GTS](http://gts.sourceforge.net/) and a few others that are required for visualising large(ish) graphs, so I am using a graphviz that I compile with all the custom features and then the python stuff links against that one.
